sklearn/ensemble/forest.py

```diff
@@ -565,31 +579,30 @@ class in a leaf.
         # Assign chunk of trees to jobs
         n_jobs, _, _ = _partition_estimators(self.n_estimators, self.n_jobs)
 
-        # Parallel loop
-        all_proba = Parallel(n_jobs=n_jobs, verbose=self.verbose,
-                             backend="threading")(
-            delayed(parallel_helper)(e, 'predict_proba', X,
-                                      check_input=False)
-            for e in self.estimators_)
-
-        # Reduce
-        proba = all_proba[0]
-
+        # avoid storing the output of every estimator by summing them in `out`
         if self.n_outputs_ == 1:
-            for j in range(1, len(all_proba)):
-                proba += all_proba[j]
+            proba = np.zeros((X.shape[0], self.n_classes_), dtype=np.float64)
+
+            # Parallel loop
+            Parallel(n_jobs=n_jobs, verbose=self.verbose, backend="threading")(
+                delayed(_run_estimator)(e.predict_proba, X, proba)
+                for e in self.estimators_)
 
             proba /= len(self.estimators_)
+            return proba
 
         else:
-            for j in range(1, len(all_proba)):
-                for k in range(self.n_outputs_):
-                    proba[k] += all_proba[j][k]
+            all_proba = [np.zeros((X.shape[0], j), dtype=np.float64)
+                         for j in self.n_classes_]
```

Comment:
if you do `np.atleast_1d` around `self.n_classes_` then you can generalise this to the 1 output case quite easily